src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionFactory.java

```diff
@@ -167,11 +170,14 @@ private LazyQueryCache(int capacity) {
         public QueryCache get() {
             QueryCache cache = this.cache;
             if (cache == null) {
-                synchronized (this) {
+                try {
+                    lock.lock();
```

Comment:
How about moving `lock#lock()` before try statement? See the example at [here](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/ReentrantLock.html).

Reply:
Yep, all `lock()` should be preceded by `try`

Reply:
Oh, I've just fixed it.
Thanks :D